File: app/src/main/java/lemuel/lemubit/com/biometricattendance/util/TimeHelper.java
package lemuel.lemubit.com.biometricattendance.util;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class TimeHelper {
    public static String getCurrentTime() {
        SimpleDateFormat currentTime = new SimpleDateFormat("hh:mm:ss a", Locale.UK);
        return currentTime.format(new Date());
    }

    public static String getCurrentDate() {
        SimpleDateFormat currentDate = new SimpleDateFormat("dd/MM/yyyy", Locale.UK);
        return currentDate.format(new Date());
    }

    public static Boolean isPassedDeadline(String registeredTime) throws ParseException {
        SimpleDateFormat parser = new SimpleDateFormat("hh:mm:ss a", Locale.UK);
        Date mCurrentTime = parser.parse(registeredTime);
        Date mDeadline = parser.parse("08:14:59 am");

        return mCurrentTime.after(mDeadline);
    }
}

File: app/src/main/java/lemuel/lemubit/com/biometricattendance/util/TimeHelper.kt
package lemuel.lemubit.com.biometricattendance.util

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

object TimeHelper {
    val currentTime: String
        get() {
            val currentTime = SimpleDateFormat("hh:mm:ss a", Locale.UK)
            return currentTime.format(Date())
        }

    val currentDate: String
        get() {
            val currentDate = SimpleDateFormat("dd/MM/yyyy", Locale.UK)
            return currentDate.format(Date())
        }

    @Throws(ParseException::class)
    fun isPassedDeadline(registeredTime: String): Boolean {
        val parser = SimpleDateFormat("hh:mm:ss a", Locale.UK)
        val mCurrentTime = parser.parse(registeredTime)
        val mDeadline = parser.parse("08:14:59 am")

        return mCurrentTime.after(mDeadline)
    }

     fun formatDate(year: Int, month: Int, day: Int): String {
        val cal = Calendar.getInstance()
        cal.set(year, month, day)
        val format1 = SimpleDateFormat("dd/MM/yyyy", Locale.UK)
        val formatted = format1.format(cal.time)

        return formatted
    }
}
